src/einsteinpy/metric/schwarzschild.py

```diff
@@ -1,199 +1,146 @@
+        """
+        Returns f_vec for Schwarzschild Metric
+        To be used in solving for Geodesics
 
         Parameters
         ----------
-        start_lambda : float
-            Starting lambda, defaults to 0.0, (lambda ~= t)
-        stop_on_singularity : bool
-            Whether to stop further computation on reaching singularity, defaults to True
-        OdeMethodKwargs : dict
-            Kwargs to be supplied to the ODESolver, defaults to {'stepsize': 1e-3}
-            Dictionary with key 'stepsize' along with an float value is expected.
-        return_cartesian : bool
-            True if coordinates and velocities are required in cartesian coordinates(SI units), defaults to Falsed
-
-        Yields
-        ------
-        float
-            proper time
-        ~numpy.ndarray
-            array of [t, x1, x2, x3, velocity_of_time, v1, v2, v3] for each proper time(lambda).
+        lambda_ : float
+            Parameterizes current integration step
+            Used by ODE Solver
+
+        vec : numpy.array
+            Length-8 Vector, containing 4-Position & 4-Velocity
+
+        Returns
+        -------
+        ~numpy.array
+            f_vec for Schwarzschild Metric
+            Numpy array of shape (8)
 
         """
-        ODE = RK45(
-            fun=self.f_vec,
-            t0=start_lambda,
-            y0=self.initial_vec,
-            t_bound=1e300,
-            **OdeMethodKwargs
+        chl = self.christoffels(vec[:4])
+        vals = np.zeros(shape=vec.shape, dtype=vec.dtype)
+
+        vals[:4] = vec[4:]
+        vals[4] = -2 * chl[0, 0, 1] * vec[4] * vec[5]
+        vals[5] = -1 * (
+            chl[1, 0, 0] * (vec[4] ** 2)
+            + chl[1, 1, 1] * (vec[5] ** 2)
+            + chl[1, 2, 2] * (vec[6] ** 2)
+            + chl[1, 3, 3] * (vec[7] ** 2)
         )
+        vals[6] = -2 * chl[2, 2, 1] * vec[6] * vec[5] - 1 * chl[2, 3, 3] * (vec[7] ** 2)
+        vals[7] = -2 * (chl[3, 3, 1] * vec[7] * vec[5] + chl[3, 3, 2] * vec[7] * vec[6])
+
+        return vals
+
+    # time_velocity moved to `coordinates.utils` as v_t()
+    # calculate_trajectory moved to `geodesic`
 
-        crossed_event_horizon = False
-        _scr = self.scr.value * 1.001
-
-        while True:
-            if not return_cartesian:
-                yield ODE.t, ODE.y
-            else:
-                v = ODE.y
-                si_vals = SphericalConversion(
-                    v[1], v[2], v[3], v[5], v[6], v[7]
-                ).convert_cartesian()
-                yield ODE.t, np.hstack((v[0], si_vals[:3], v[4], si_vals[3:]))
-            ODE.step()
-            if (not crossed_event_horizon) and (ODE.y[1] <= _scr):
-                warnings.warn("particle reached Schwarzchild Radius. ", RuntimeWarning)
-                if stop_on_singularity:
-                    break
-                else:
-                    crossed_event_horizon = True
+    # Hiding unrelated methods
+    em_potential_covariant = BaseMetric._private
```

Comment:
Again, not satisfied with the design here.

Reply:
Also, I have to say that the name of the variables makes (at least to me) a bit difficult the understanding of what is actually doing the code